src/brokers/broker-pods/PodsList.container.tsx

```diff
@@ -36,11 +36,22 @@ export const PodsContainer: FC = () => {
   });
 
   const filterBrokerPods = (pods: K8sResourceCommon[], brokerName: string) => {
-    return pods.filter((pod) =>
-      pod.metadata?.ownerReferences?.some(
-        (ref) => ref.name.startsWith(brokerName) && ref.kind === 'StatefulSet',
-      ),
-    );
+    return pods.filter((pod) => {
+      if (!pod.metadata?.ownerReferences) {
+        return false;
+      }
+      return pod.metadata.ownerReferences.some((ref) => {
+        if (ref.kind !== 'StatefulSet') {
+          return false;
+        }
+        const refName = ref.name;
+        const refNameWithoutSuffix = refName.endsWith('-ss')
+          ? refName.slice(0, -3)
+          : refName;
+        const isMatch = refNameWithoutSuffix === brokerName;
+        return isMatch;
+      });
+    });
```

Comment:
While this seems good and looks to work quite well, I think we can improve it by using a regex, something along those lines:

```typescript
    return pods.filter((pod) =>
      pod.metadata?.ownerReferences?.some(
        (ref) => ref.name.match(`^${brokerName}-ss[0-9]$?`) && ref.kind === 'StatefulSet',
      ),
    );
```

Reply:
Pods name follows the pattern <brokerName>-ss-<oridinal number> . The regex need to have a unit test.

Reply:
> Pods name follows the pattern `[brokerName]-ss-[oridinal number]` . So the owner ref name (statefulset) is `[brokerName]-ss`

Hello @gaohoward , yes that's correct the statefulset has pattern of `[brokerName]-ss` i updated this same approach in my changes.